pghoard/webserver.py

```diff
@@ -353,34 +390,46 @@ def get_wal_or_timeline_file(self, site, filename, filetype):
         if os.path.exists(xlog_path):
             self.server.log.info("Requested %r, found it in pg_xlog directory as: %r, returning directly",
                                  filename, xlog_path)
-            success = False
-            try:
-                self._save_and_verify_restored_file(filetype, filename, xlog_path, target_path)
-                success = True
-            except (ValueError, HttpResponse) as ex:
+            ex = self._try_save_and_verify_restored_file(filetype, filename, xlog_path, target_path, unlink=False)
+            if ex:
                 self.server.log.warning("Found file: %r but it was invalid: %s", xlog_path, ex)
-            if success:
+            else:
                 raise HttpResponse(status=201)
 
-        prefetch_n = self.server.config["restore_prefetch"]
-        prefetch = []
-        if filetype == "timeline":
-            tli_num = int(filename.replace(".history", ""), 16)
-            for _ in range(prefetch_n):
-                tli_num += 1
-                prefetch.append("{:08X}.history".format(tli_num))
-        elif filetype == "xlog":
-            xlog_num = int(filename, 16)
-            for _ in range(prefetch_n):
-                if xlog_num & 0xFF == 0xFF:
-                    xlog_num += 0xFFFFFF00
-                xlog_num += 1
-                prefetch.append("{:024X}".format(xlog_num))
+        key = self._make_file_key(site, filetype, filename)
+        with suppress(ValueError):
+            self.server.prefetch_404.remove(key)
+        self._create_fetch_operation(key, site, filetype, filename, max_age=5, suppress_error=False)
+        self._create_prefetch_operations(site, filetype, filename)
 
-        with self._prefetch(site, filetype, prefetch):
-            self._transfer_agent_op(site, filename, filetype, "DOWNLOAD", target_path=target_path)
-
-        raise HttpResponse(status=201)
+        last_schedule_call = time.monotonic()
+        start_time = time.monotonic()
+        retries = 2
+        while time.monotonic() - start_time <= 30:
```

Comment:
explicit parenthesis for added readability: `(time.monotonic() - start_time) <= 30`